src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -194,11 +189,14 @@ private void initialize(Context appContext, String authority, ITokenCacheStore t
             throw new UnsupportedOperationException("Local cache is not supported for broker usage");
         }
         mContext = appContext;
-        mConnectionService = new DefaultConnectionService(mContext);
         checkInternetPermission();
         mAuthority = extractAuthority(authority);
         mValidateAuthority = validateAuthority;
         mTokenCacheStore = tokenCacheStore;
+        
+        if (mTokenCacheStore != null) {
+            mTokenCacheAccessor = new TokenCacheAccessor(mTokenCacheStore, mAuthority);
```

Comment:
My intention here is if token cache store is null, i shouldn't even interact with tokencacheaccessor

---
In reply to: [63266101](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/611#discussion_r63266101) [](ancestors = 63266101)